### PR TITLE
Update wheelspeed input to FPB_Measurements

### PR DESCRIPTION
**Pull request by joaomsbernardino**
This MR updates the the ros driver to the newly supported fpb measurements message definitions to input wheelspeed to the sensor. This allows to generalize the velocity measurements to 3D instead of the currently supported 1D.
As a side effect, the usage of the RAWDMI message is also removed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -239,48 +239,16 @@ _Please note that the corresponding messages also has to be selected on the Fixp
...
+
+Regarding the number of sensors that are supported, there are currently 2 options:
+
+-   Option 1: Only one vehicle speed, corresponding to the sensor RC.
+-   Option 2: 4 Values of 4 wheels, in the order of FR, FL, RR, RL
+
+Note that if your platform has a different number of wheels, (e.g. 2), it is possible to send to the 4 wheelspeed sensor values to the sensor. This is merely thought as a generic way to get values into the sensor. You can then choose, in the normal sensor configuration, which sensors are available, respectively, which ones to use in the fusion engine. 
+
+Internally, upon arriving to the ros driver, wheelspeed measurements are converted into a FP_B-MEASUREMENTS message and sent via the TCP or serial interface to the Vision-RTK2, where they will be further processed. For more details regarding the definition FP_B-MEASUREMENTS message, please refer to the integration manual.
```
Comment (souryavarenya) [inline README.md:251]: Please link this:

https://docs.fixposition.com/fd/fp_b-measurements

File: README.md [hunk 2]
```diff
@@ -239,48 +239,16 @@ _Please note that the corresponding messages also has to be selected on the Fixp
...
-
-Measurement types (`dmi1_type`, `dmi2_type`, `dmi3_type` and `dmi4_type`):
-
-| Value | Description             |
-| :---: | ----------------------- |
-|  `0`  | Linear velocity (speed) |
-|  `1`  | Angular velocity        |
-
-Note: _Currently the wheelspeed input through the ROS driver is only supported in the TCP mode_
+The fp_ros_driver support inputing a Speed msg (`msg/Speed.msg`) through the `/fixposition/speed` topic. The Speed msg is defined as a vector of WheelSensor msgs (`msg/WheelSensor.msg`). This message is, in turn, defined as a vector of 32bit integers. You are free to fill it with either 1, 2 or 3 values, corresponding to the x, y, and z axis of your wheelspeed sensor. These velocity values should be in [mm/s], in order to have enough precision in an integer format.
```
Comment (souryavarenya) [inline README.md:242]:
```suggestion
The fp_ros_driver supports inputting a Speed msg (`msg/Speed.msg`) through the `/fixposition/speed` topic. The Speed msg is defined as a vector of WheelSensor msgs (`msg/WheelSensor.msg`). This message is, in turn, defined as a vector of 32-bit integers. You are free to fill it with either 1, 2, or 3 values, corresponding to the x, y, and z axis of your wheelspeed sensor. These velocity values should be in [mm/s], to have enough precision in an integer format.
```

grammarly

File: README.md [hunk 3]
```diff
@@ -239,48 +239,16 @@ _Please note that the corresponding messages also has to be selected on the Fixp
...
+
+Regarding the number of sensors that are supported, there are currently 2 options:
+
+-   Option 1: Only one vehicle speed, corresponding to the sensor RC.
+-   Option 2: 4 Values of 4 wheels, in the order of FR, FL, RR, RL
+
+Note that if your platform has a different number of wheels, (e.g. 2), it is possible to send to the 4 wheelspeed sensor values to the sensor. This is merely thought as a generic way to get values into the sensor. You can then choose, in the normal sensor configuration, which sensors are available, respectively, which ones to use in the fusion engine.
```
Comment (souryavarenya) [inline README.md:249]: Refactor this to be flexible and deterministic as per our discussion.

File: fixposition_driver_ros1/msg/Speed.msg [hunk 4]
```diff
@@ -13,9 +13,8 @@
 #
 ####################################################################################################
```
Comment (souryavarenya) [inline fixposition_driver_ros1/msg/Speed.msg:15]: Can/Should we support user timestamping?

File: fixposition_driver_ros1/msg/WheelSensor.msg [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+#
+#
+####################################################################################################
+
+# Velocity values in [mm/s] as integer 32bit
+# Options:
+# Option 1: One value, will be input in the x axis
+# Option 2: Two values, will be input in the x and y axis
+# Option 2: Three values, will be input in the x, y, and z axis
+int32[] speeds
```
Comment (souryavarenya) [inline fixposition_driver_ros1/msg/WheelSensor.msg:21]: These documentation options look a little loose to me. We can IMO be a little more transparent to the FP_B-MEASUREMENTS definition. We can't for example just specify `y` velocity if the sensor is mounted sideways.

Can we switch this to something like:

```suggestion
int32 vx
int32 vy
int32 vz

bool vx_valid
bool vy_valid
bool vz_valid
```
For example: https://docs.ros.org/en/melodic/api/sensor_msgs/html/msg/BatteryState.html

File: fixposition_driver_ros2/msg/Speed.msg [hunk 6]
```diff
@@ -15,4 +15,4 @@
 # Option 1: One vehicle speed, of sensor RC
 # Option 2: One vehicle speed, of sensor RC and the yaw rate of the vehicle
 # Option 3: Fill in 4 Values of 4 wheels, in the order of FR, FL, RR, RL
-int32[] speeds
+fixposition_driver_ros2/WheelSensor[] sensors
```
Comment (souryavarenya) [inline fixposition_driver_ros2/msg/Speed.msg:18]:
```suggestion
fixposition_driver_ros2/WheelSensor[] sensors

```

Comment (souryavarenya) [reply fixposition_driver_ros2/msg/Speed.msg:18]: EOF

File: fixposition_odometry_converter/README.md
Comment (souryavarenya) [inline fixposition_odometry_converter/README.md:1]: Rename this folder to `fixposition_odometry_converter_ros1`?

File: fixposition_odometry_converter/launch/odom_converter.yaml [hunk 7]
```diff
@@ -2,4 +2,4 @@ topic_type: "Twist" # Twist, TwistWithCov or Odometry
 input_topic: "/robot/odom" # Original topic name
 fixposition_speed_topic: "/fixposition/speed" # Fixposition topic name, to be subscribed by the fixposition_driver
 multiplicative_factor: 1000 # To convert from the original unit of measurement to mm/s
-use_angular: false # If the angular velocity of the speed message is filled
+use_dimensions: 1 # 1 to use x only, 2 to use x and y, 3 to use x y z
```
Comment (souryavarenya) [inline fixposition_odometry_converter/launch/odom_converter.yaml:5]:
```suggestion
use_dimensions: 1 # 1 to use x only, 2 to use x and y, 3 to use x y z

```

EOF

File: fixposition_odometry_converter_ros2/launch/odom_converter.yaml [hunk 8]
```diff
@@ -1,7 +1,7 @@
 /**:
   ros__parameters:
-    topic_type: "Odometry" # Twist, TwistWithCov or Odometry
+    topic_type: "Twist" # Twist, TwistWithCov or Odometry
     input_topic: "/robot/odom" # Original topic name
     fixposition_speed_topic: "/fixposition/speed" # Fixposition topic name, to be subscribed by the fixposition_driver
     multiplicative_factor: 1000 # To convert from the original unit of measurement to mm/s
-    use_angular: false # If the angular velocity of the speed message is filled
+    use_dimensions: 1 # 1 to use x only, 2 to use x and y, 3 to use x y z
```
Comment (souryavarenya) [inline fixposition_odometry_converter_ros2/launch/odom_converter.yaml:7]:
```suggestion
    use_dimensions: 1 # 1 to use x only, 2 to use x and y, 3 to use x y z

```

EOF

Comment (souryavarenya) [inline fixposition_odometry_converter_ros2/launch/odom_converter.yaml:7]: I don't see the need for any of this configurability in this node, except for the input topic. 

Lower configurability = fewer customer mistakes 

- type can be auto-deduced
- fixposition_speed_topic is constant
- Multiplicative factor is not needed (can be constant to 1000)
- use_dimensions should ideally be `x, y, z` if we're converting from standard ros messages

File: fixposition_driver_ros1/msg/Speed.msg (hunk 4 above)
Comment (joaomsbernardino) [reply fixposition_driver_ros1/msg/Speed.msg:15]: That is the plan in the semi near future, and the fpb measurements has the infrastructure for it. Currently however we dont support it internally.

File: fixposition_driver_ros1/msg/Speed.msg [hunk 9]
```diff
@@ -13,9 +13,5 @@
...
 
-# Velocity values in [mm/s] (or [mrad/s]) as integer 32bit
-# 2 Options:
-# Option 1: One vehicle speed, of sensor RC
-# Option 2: One vehicle speed, of sensor RC and the yaw rate of the vehicle
-# Option 3: Fill in 4 Values of 4 wheels, in the order of FR, FL, RR, RL
-int32[] speeds
+# Velocity values for one or several WheelSensors
```
Comment (souryavarenya) [inline fixposition_driver_ros1/msg/Speed.msg:16]:
```suggestion
Header header

# Velocity values for one or several WheelSensors
```

**Pull request merged**